lib/Serialization/SerializeSIL.cpp

```diff
@@ -675,6 +681,8 @@ SILSerializer::writeKeyPathPatternComponent(
 }
 
 void SILSerializer::writeSILInstruction(const SILInstruction &SI) {
+  TraceSILNode trace(&SI, currentSILNode);
```

Comment:
This seems needlessly stateful. I know it hits TLS, but is it really too expensive to use a PrettyStackTraceSILNode every time?

Reply:
No, people are just bad at remembering that PrettyStackTrace exists. Let's go with the normal idiom for now since it means less shared state.

Reply:
I agree, the small TLS overhead should never be noticeable, even in release builds.